soc/arm/silabs_exx32/common/soc.c

```diff
@@ -70,6 +71,25 @@ static ALWAYS_INLINE void clkInit(void)
 #endif
 }
 
+static ALWAYS_INLINE void dcDcInit(void)
```

Comment:
I would code this function differently. Here is my proposal:
```
#ifdef CONFIG_SOC_GECKO_EMU_DCDC
static ALWAYS_INLINE void dcdc_init(void)
{
#if defined(CONFIG_SOC_GECKO_EMU_DCDC_MODE_UNCONFIGURED)
	/* Nothing to do, leave DC/DC converter in unconfigured, safe state. */
#elif defined(CONFIG_SOC_GECKO_EMU_DCDC_MODE_ON) || defined(CONFIG_SOC_GECKO_EMU_DCDC_MODE_BYPASS)
	EMU_DCDCInit_TypeDef init_cfg = EMU_DCDCINIT_DEFAULT;
#if defined(CONFIG_SOC_GECKO_EMU_DCDC_MODE_BYPASS)
	init_cfg.dcdcMode = emuDcdcMode_Bypass;
#endif
	EMU_DCDCInit(&init_cfg);
#elif defined(CONFIG_SOC_GECKO_EMU_DCDC_MODE_OFF)
	EMU_DCDCPowerOff();
#else
#error "Unsupported power configuration mode of the on chip DC/DC converter."
#endif
}
#endif
```
It should be called like this
```
#ifdef CONFIG_SOC_GECKO_EMU_DCDC
	/*
	 * Initialize DCDC before attempting to set clock, HFXO for example
	 * requires this
	 */
	dcdc_init();
#endif
```
Dependent code, e.g. Kconfig options should be updated accordingly.

Rationale:
- function name should follow Zephyr [Coding Style](https://docs.zephyrproject.org/latest/contribute/contribute_guidelines.html#coding-style). I understand that we already broke this rule and have names like clkInit, oldLevel. I reckon we should clean it up, albeit in a separate commit.
- surrounding call to `dcdc_init()` in the `#ifdef` is ugly, however makes it obvious for person reading `silabs_exx32_init()` function code that this part of the code is optional. Someone who's working with SiLabs MCU without DC/DC converter should not see a call to dcdc_init() function, even it the function is empty.
- let's not use `SOC_DCDC_GECKO` convention for SoC config options even if this is a style used by all the drivers e.g. `UART_GECKO_0`. In case of SoC options it makes them very confusing. Better convention to follow is `SOC_GECKO_<FEATURE>`. The addition of `_EMU` is optional but should make things easier when we add power management code which will also depend on the EMU module. You may remove the `_EMU` if you think it makes the name too long.